### PR TITLE
detect.py: Fix KeyError if using MinGW and LTO

### LINKED ISSUE #12433: Error when building on Windows with MinGW with LTO enabled
**Operating system or device, Godot version, GPU Model and driver (if graphics related):** Windows 10 Fall Creators Update, Godot Git https://github.com/godotengine/godot/commit/ad85224ebcf7bbad5565457d355c429b02190c58

**Issue description:** Using MinGW-w64 from [MSYS2](http://www.msys2.org/), the build fails when `use_lto=yes` is used. However, it works when using `use_lto=no` (or when leaving it out, which defaults to `no`).

See the error log below (using Python 2.7.13):

```
scons: warning: No version of Visual Studio compiler found - C/C++ compilers most likely not set correctly
File "C:\Users\hugol\Documents\Git\godotengine\godot\SConstruct", line 4, in <module>
MSVC not detected (no VCINSTALLDIR environment variable), attempting MinGW.
KeyError: 'use_llvm':
  File "C:\Users\hugol\Documents\Git\godotengine\godot\SConstruct", line 300:
    detect.configure(env)
  File "./platform/windows\detect.py", line 267:
    if not env['use_llvm'] and env.GetOption("num_jobs") > 1:
  File "C:\Python27\Scripts\..\Lib\site-packages\scons-2.5.1\SCons\Environment.py", line 410:
    return self._dict[key]
```

**Steps to reproduce:** Build Godot with `use_lto=yes` with a MinGW-w64 compiler in the `PATH`.

### PR DESCRIPTION
**Pull request by mhilbrunner**
Fixes #12433.

### CODE REVIEW

File: platform/windows/detect.py [hunk 1]
```diff
@@ -264,7 +264,7 @@ def configure(env):
 
         if env['use_lto']:
             env.Append(CCFLAGS=['-flto'])
-            if not env['use_llvm'] and env.GetOption("num_jobs") > 1:
+            if not 'use_llvm' in env and env.GetOption("num_jobs") > 1:
```
Comment (akien-mga) [inline platform/windows/detect.py]: `use_llvm` just does not make sense here, it's a wrong copy paste from the x11 file.
It should simply be:
```
env.Append(LINKFLAGS=['-flto=' + str(env.GetOption("num_jobs"))])
```
in place of the whole `if`. (provided `-flto=1` works)

Comment (mhilbrunner) [reply platform/windows/detect.py]: @akien-mga If my Google-fu didn't fail me, GCC only supports "-flto", not "-flto=<num>"? In which case, why not leave it as it is, as it should now work? And will continue working if yout decide to add support for LLVM on Windows some day. Or only use `env.Append(LINKFLAGS=['-flto'])` I guess.

Comment (mhilbrunner) [reply platform/windows/detect.py]: `-flto=1` seems to work on GCC.

**Comment by mhilbrunner** [comment]
@akien-mga Done!

**Pull request merged**